include/ignition/rendering/LidarVisual.hh

```diff
@@ -0,0 +1,165 @@
+      /// \return The maximum horizontal angle value of the lidar visual
+      public: virtual double MaxHorizontalAngle() const = 0;
+
+      /// \brief Set number of vertical rays
+      /// \param[in] _verticalRayCount Number of vertical rays
+      public: virtual void SetVerticalRayCount(
+                  const unsigned int _verticalRayCount) = 0;
+
+      /// \brief Get number of vertical rays
+      /// \return The number of vertical rays
+      public: virtual unsigned int VerticalRayCount() const = 0;
+
+      /// \brief Set number of horizontal rays
+      /// \return _horizontalRayCount Number of vertical rays
+      public: virtual void SetHorizontalRayCount(
+                  const unsigned int _horizontalRayCount) = 0;
+
+      /// \brief Get number of horizontal rays
+      /// \return The number of horizontal rays
+      public: virtual unsigned int HorizontalRayCount() const = 0;
+
+      /// \brief Set minimum range of lidar rays
+      /// \param[in] _minRange The minimum range for lidar rays
+      public: virtual void SetMinRange(const double _minRange) = 0;
+
+      /// \brief Get minimum range of lidar rays
+      /// \return The minimum range for lidar rays
+      public: virtual double MinRange() const = 0;
+
+      /// \brief Set maximum range of lidar rays
+      /// \param[in] _maxRange The maximum range for lidar rays
+      public: virtual void SetMaxRange(const double _maxRange) = 0;
+
+      /// \brief Get maximum range of lidar rays
+      /// \return The maximum range for lidar rays
+      public: virtual double MaxRange() const = 0;
+
+      /// \brief Set vertical step angle
+      /// \param[in] _verticalAngleStep Vertical angle between two rays
+      public: virtual void SetVerticalAngleStep(
+                  const double _verticalAngleStep) = 0;
+
+      /// \brief Get vertical step angle
+      /// \return The difference in vertical angle between two rays
+      public: virtual double VerticalAngleStep() const = 0;
+
+      /// \brief Set horizontal step angle
+      /// \param[in] _horizontalAngleStep Horizontal angle between two rays
+      public: virtual void SetHorizontalAngleStep(
+                  const double _horizontalAngleStep) = 0;
+
+      /// \brief Get horizontal step angle
+      /// \return The difference in horizontal angle between two rays
+      public: virtual double HorizontalAngleStep() const = 0;
+
+      /// \brief Set offset of visual
+      /// \param[in] _offset The offset of the lidar rays from visual origin
+      public: virtual void SetOffset(
+                  const ignition::math::Pose3d _offset) = 0;
+
+      /// \brief Get offset of visual
+      /// \return The offset of the lidar rays from visual origin
+      public: virtual ignition::math::Pose3d  Offset() const = 0;
+
+      /// \brief Get number of points in laser data
+      /// \return The number of points in the laser data
+      public: virtual unsigned int GetPointCount() const = 0;
```

Comment:
we follow google coding style guidelines which don't use `Get` prefix in getter functions. So `GetPointCount()` -> `PointCount()`